# Project 1 MCS 472 

Abraham Morales Arroyo 


I am going to have two different files, where the first file containes the Minimum Average, Max Average, and Average Data for December and January, respectfully. This file will be caled ``Classtoday.txt``, and it is from the year 1922- 1932. Then there is another file called ``Classtoday2.txt`` Where it has the same data, but instead it is from the years 2012-2022. 


In [14]:
using DelimitedFiles

In [25]:
A = readdlm("Classtoday.txt")

23×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 1922        21.18     31.81     26.5     5.65     22.85     14.25
 1923        22.3      32.9      27.64   11.2      25.65     18.43
 1924        22.34     32.9      27.6    -3.25     25.65      5.65
 1925        -0.36     14.18      6.91    9.9      23.65     16.78
 1926        12.91     27.91     20.41   14.95     28.95     21.95
 1927        14.0      27.82     20.91    7.75     22.6      15.18
 1928        18.36     30.55     24.46   17.3      30.9      24.1
 1929        16.91     31.55     24.23   -2.95     13.95      5.5
 1930        13.73     28.0      20.86    9.0      22.3      15.65
 1931        31.818    38.46     35.14   17.8      31.05     24.43
 1932        25.18     37.09     31.14   22.5      33.0      27.75
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.

# The Headers here are abreviations: 

 AMiD= Average Minimum Temperature in December 

 AMAD= Average Maximum Temperature in December

 AD= Average December Temperature

 AMiJ= Average Minimum Temperature in January 

 AMAJ= Average Maximum Temperature in January

AJ= Average January Temperature


For the previous instruction to work, the file ``Classtoday.txt`` has to be on file.

To make the notebook self contained, the string representation of the file will be copied and pasted into a cell.

In [26]:
strA = string(A)

"Any[\"Year\" \"AMiD\" \"AMAD\" \"AD\" \"AMiJ\" \"AMaJ\" \"AJ\"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 " ⋯ 387 bytes ⋯ ".82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]"

In [27]:
print(strA)

Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 7.75 22.6 15.18; 1928 18.36 30.55 24.46 17.3 30.9 24.1; 1929 16.91 31.55 24.23 -2.95 13.95 5.5; 1930 13.73 28.0 20.86 9.0 22.3 15.65; 1931 31.818 38.46 35.14 17.8 31.05 24.43; 1932 25.18 37.09 31.14 22.5 33.0 27.75; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

In [29]:
B=Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 7.75 22.6 15.18; 1928 18.36 30.55 24.46 17.3 30.9 24.1; 1929 16.91 31.55 24.23 -2.95 13.95 5.5; 1930 13.73 28.0 20.86 9.0 22.3 15.65; 1931 31.818 38.46 35.14 17.8 31.05 24.43; 1932 25.18 37.09 31.14 22.5 33.0 27.75; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

23×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 1922        21.18     31.81     26.5     5.65     22.85     14.25
 1923        22.3      32.9      27.64   11.2      25.65     18.43
 1924        22.34     32.9      27.6    -3.25     25.65      5.65
 1925        -0.36     14.18      6.91    9.9      23.65     16.78
 1926        12.91     27.91     20.41   14.95     28.95     21.95
 1927        14.0      27.82     20.91    7.75     22.6      15.18
 1928        18.36     30.55     24.46   17.3      30.9      24.1
 1929        16.91     31.55     24.23   -2.95     13.95      5.5
 1930        13.73     28.0      20.86    9.0      22.3      15.65
 1931        31.818    38.46     35.14   17.8      31.05     24.43
 1932        25.18     37.09     31.14   22.5      33.0      27.75
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.

The instructions in the notebook will continue to work with ``B`` so everything will work even if the file was not present.


# Now we are going to have the rest of the data, From the years 2012-2022, be read and then stored into a matrix so it can be used for this project: 

In [140]:
C = readdlm("Classtoday2.txt")

12×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.45     20.75     11.6
 2015        25.73     34.82     30.27    7.5      21.45     14.48
 2016        22.82     35.73     29.27    8.6      23.3      15.95
 2017        -3.28     13.36      5.05   12.95     25.4      19.18
 2018        22.27     34.09     28.68    7.4      23.95     15.68
 2019        28.18     39.09     33.64    8.6      23.3      15.95
 2020        15.64     31.55     23.59   17.75     32.75     25.25
 2021        17.36     33.36     25.36   20.7      30.9      25.8
 2022         9.27     26.09     17.68    6.8      24.15     15.48

In [141]:
strC = string(C)

"Any[\"Year\" \"AMiD\" \"AMAD\" \"AD\" \"AMiJ\" \"AMaJ\" \"AJ\"; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]"

In [142]:
D=Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

12×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.45     20.75     11.6
 2015        25.73     34.82     30.27    7.5      21.45     14.48
 2016        22.82     35.73     29.27    8.6      23.3      15.95
 2017        -3.28     13.36      5.05   12.95     25.4      19.18
 2018        22.27     34.09     28.68    7.4      23.95     15.68
 2019        28.18     39.09     33.64    8.6      23.3      15.95
 2020        15.64     31.55     23.59   17.75     32.75     25.25
 2021        17.36     33.36     25.36   20.7      30.9      25.8
 2022         9.27     26.09     17.68    6.8      24.15     15.48

# Let us take the average of minimum temperatures of December for the first decade, 1922- 1932

In [36]:
using Statistics

here we are making the selection of the month that we want 

In [37]:
B[1,2]

"AMiD"

In [50]:
mean(B[2:12,2]) # column 2, rows 3 and 4 just as an example

18.033454545454546

 As we can see the average temperature for December 20th - 31st from 1922-1932 is about ``18.03`` (rounded) 

# Now we are going to be comparing this to the minimum temperature of Dec 20th- 31st from 2012-2022: 

Making the selection of the month: 

In [144]:
D[1,2]

"AMiD"

now we are going to be taking the average: 

In [146]:
mean(D[2:12,2]) # Remember that for the years 2012-2022 is going to be stored in the matrix named ``D``

16.056818181818183

Here we see that we got the average of minimum average temperature for December, 2012-2022, to be ``16.06``, rounded

# Outcome: 

When we compare the average temperature from 1922-1932, Average Min December: ``18.03``. To the data 2012-2022: ``16.06`` we see that for the years 2012-2022 the average temperature is much lower than the average temperature from 1922-1932.  

There is a difference of: 

In [147]:
18.03-16.06

1.9700000000000024

The difference from the years 1922-1932 to 2012-2022 is: ``1.9700000000000024``, meaning it has become colder.

# We will now do the same but for the average Max temperature for December 1922-1932

# This below is legacy code: DO NOT MESS WITH IT!!!

In [143]:
D[2:12,2]# this gives the 'first decade'

11-element Vector{Any}:
 13.0
  4.09
 21.545
 25.73
 22.82
 -3.28
 22.27
 28.18
 15.64
 17.36
  9.27

In [49]:
B[2:12,1]# this gives the years 

11-element Vector{Any}:
 1922
 1923
 1924
 1925
 1926
 1927
 1928
 1929
 1930
 1931
 1932